Topic 1/src/point6/Factory.java

```diff
@@ -0,0 +1,25 @@
+package point6;
+
+public class Factory {
+	public AbstractPayment getPayment(String name, double amount, ShoppingCart shoppingCart, MailList mailList) throws Exception {
```

Comment:
[Medium] Although it is ok to create a factory to return the proper payment type, not always it's a good option.
In this case, you are sending a lot of information and not all of it is used for all payment types (*cash* only use the amoung to pay for example).
Perhaps in this case it would be better not to have a *factory* and create the proper instance when you need it.